Improve rosetta docs (#727)
o1-labs/docs2: docs/exchange-operators/rosetta/samples/index.mdx

---
title: Code Samples
description: Inspect integration code samples
---

:::info

The code samples in this section are tested with Python 3.9

:::

The following series of guides demonstrate how to perform common Rosetta API use-cases. 

All code samples are written in Python and require the `requests` library to be installed on your Python environment.

This can be done with a following command:

    pip3 install requests

> Also for using `pprint_json` function, you will need `jq` installed on your machine. On Linux machines, it is installed by default. For macOS, you can install it with brew package manager:

Reviewer: I don't think this is the case. I have just checked a 22.04 LTS Ubuntu installation and it does not include `jq` by default. Arch Linux that I use does not ship with it either. It needs to be installed manually.

Author: removed jq

    brew install jq

The source code for all code samples is available on the official [Mina Documentation Github page](https://github.com/o1-labs/docs2/tree/main/examples/exchange-operators/rosetta_samples.py) and should be used only for demonstration purposes.

# Set up sample project

> Let's start with necessary imports and define constants:

Reviewer 2: ```suggestion
Start with the required imports and define constants:
```

```python
import json
from os import system
from subprocess import run, PIPE
from time import sleep

from requests import post

BASE_URL = "http://localhost:3087"
SIGNER_PATH = 'path/to/signer.exe'
MAINNET_NETWORK_IDENTIFIER = {
    "network_identifier": {
        "blockchain": "mina",
        "network": "mainnet"
    }
}
MINA_TOKEN_ID = "1"
MINA_DECIMALS = 9
MINA_SYMBOL = "MINA"
MINA_CURVE_TYPE = "pallas"
MINA_SIGNATURE_TYPE = "schnorr_poseidon"
```

> To work with JSON requests and responses, you might want to print highlighted and indented JSON objects. You can define a simple function with the help of the `jq` utility:

Reviewer: I don't think `jq` is necessary for this. Python itself contains functionality to pretty-print JSON:
```
import json

json.dumps(j, indent=2)
```

Author: changed

> As we are going to work with json requests and responses, we may want to print highlighted and indented json objects. Let's define a simple function doing that with help of `jq` utility

Reviewer 2: ```suggestion
To work with JSON requests and responses, you might want to print highlighted and indented JSON objects. You can define a simple function with the help of the `jq` utility:
```
